dedupe/convenience.py

```diff
@@ -24,55 +24,94 @@ def consoleLabel(deduper): # pragma: no cover
+      
         examples = {'distinct' : [], 'match' : []}
 
-        for record_pair in uncertain_pairs:
-            label = ''
-            labeled = False
-
-            for pair in record_pair:
-                for field in fields:
-                    line = "%s : %s" % (field, pair[field])
-                    print(line, file=sys.stderr)
-                print(file=sys.stderr) 
-
-            print("{0}/10 positive, {1}/10 negative".format(n_match, n_distinct),
-                  file=sys.stderr)
-            print('Do these records refer to the same thing?', file=sys.stderr)
-            valid_response = False
-            while not valid_response:
+        label = ''
+        labeled = False
+
+        for pair in record_pair:
+            for field in fields:
+                line = "%s : %s" % (field, pair[field])
+                print(line, file=sys.stderr)
+            print(file=sys.stderr) 
+
+        print("{0}/10 positive, {1}/10 negative".format(n_match, n_distinct),
+                file=sys.stderr)
+        print('Do these records refer to the same thing?', file=sys.stderr)
+        valid_response = False
+        while not valid_response:
+            if has_previous:
+                print('(y)es / (n)o / (u)nsure / (f)inished / (p)revious', file=sys.stderr)
+                label = input()
+                if label in ['y', 'n', 'u', 'f', 'p']:
+                    valid_response = True
+            else:
                 print('(y)es / (n)o / (u)nsure / (f)inished', file=sys.stderr)
                 label = input()
                 if label in ['y', 'n', 'u', 'f']:
                     valid_response = True
-
-            if label == 'y' :
-                examples['match'].append(record_pair)
-                labeled = True
-            elif label == 'n' :
-                examples['distinct'].append(record_pair)
-                labeled = True
-            elif label == 'f':
-                print('Finished labeling', file=sys.stderr)
-                finished = True
-            elif label != 'u':
-                print('Nonvalid response', file=sys.stderr)
-                raise
-
-        if labeled :
-            deduper.markPairs(examples)
+                if label == 'p':
+                    print('(p)revious is not a valid answer (no record in memory)', file=sys.stderr)
+
+        if label == 'y' :
+            examples['match'].append(record_pair)
+            labeled = True
+        elif label == 'n' :
+            examples['distinct'].append(record_pair)
+            labeled = True
+        elif label == 'f':
+            print('Finished labeling', file=sys.stderr)
+            finished = True
+        elif (label == 'p') and has_previous:
+            use_previous = True
+        elif label != 'u':
+            print('Nonvalid response', file=sys.stderr)
+            raise
         
+        if labeled:
+            examples_buffer = [examples] + examples_buffer
```

Comment:
I replaced this by `buffer.insert(0, value)` in the following pull request. I can't use append here as I want to prepend the example (unless you were suggesting I invert the order of the elements in the buffer...).